ui/app/AppLayouts/Chat/views/CommunitySettingsView.qml

```diff
@@ -302,34 +321,54 @@ StatusSectionLayout {
                                                            accountName,
                                                            artworkCropRect)
                 }
-                onSignSelfDestructTransactionOpened: communityTokensStore.computeSelfDestructFee(chainId)
+                onSignSelfDestructTransactionOpened: communityTokensStore.computeSelfDestructFee(selfDestructTokensList, contractUniqueKey)
                 onRemoteSelfDestructCollectibles: {
-                    communityTokensStore.remoteSelfDestructCollectibles(selfDestructTokensList,
-                                                                        chainId,
-                                                                        accountName,
-                                                                        accountAddress)
+                    communityTokensStore.remoteSelfDestructCollectibles(root.community.id,
+                                                                        selfDestructTokensList,
+                                                                        contractUniqueKey)
                 }
                 onAirdropCollectible: root.goTo(Constants.CommunitySettingsSections.Airdrops)
 
                 Connections {
                     target: rootStore.communityTokensStore
                     function onDeployFeeUpdated(ethCurrency, fiatCurrency, errorCode) {
-                        if (errorCode === Constants.ComputeFeeErrorCode.Success || errorCode === Constants.ComputeFeeErrorCode.Balance) {
-                            let valueStr = LocaleUtils.currencyAmountToLocaleString(ethCurrency) + "(" + LocaleUtils.currencyAmountToLocaleString(fiatCurrency) + ")"
-                            mintPanel.feeText = valueStr
-                            if (errorCode === Constants.ComputeFeeErrorCode.Balance) {
-                                mintPanel.errorText = qsTr("Not enough funds to make transaction")
-                            }
-                            mintPanel.isFeeLoading = false
-                            return
-                        } else if (errorCode === Constants.ComputeFeeErrorCode.Infura) {
-                            mintPanel.errorText = qsTr("Infura error")
-                            mintPanel.isFeeLoading = true
+                        mintPanel.setFeesInfo(ethCurrency, fiatCurrency, errorCode)
+                    }
+
+                    function onSelfDestructFeeUpdated(ethCurrency, fiatCurrency, errorCode) {
+                        mintPanel.setFeesInfo(ethCurrency, fiatCurrency, errorCode)
+                    }
+
+                    function onRemoteDestructStateChanged(communityId, tokenName, status, url) {
```

Comment:
Just thinking out loud.. Could we make this `toast` preparation more global for `CommunitySettingsView` since we will need the same/similar behaviour for `burn` and `airdrop` too?? Maybe a `CommunitySettingsToastHelper` or something..